Fix several bittrex API issues

firepol opened the pull request.

deepbrook commented on bitex/interface/bittrex.py:
```diff
@@ -103,13 +102,12 @@ def cancel_order(self, *order_ids, **kwargs):
+        payload = kwargs
         if currency:
             payload = {'currency': currency}
             payload.update(kwargs)
```
Good attempt at fixing this, but better would be:

```
payload = kwargs
if currency:
    payload.update({'currency': currency})
```

firepol replied:
Even better I think it's to update kwargs and pass kwargs to the params instead of a new variable called payload. I updated the entire file accordingly.

deepbrook commented on bitex/interface/bittrex.py:
```diff
@@ -103,13 +102,12 @@ def cancel_order(self, *order_ids, **kwargs):
 
     @format_with(BittrexFormattedResponse)
-    def wallet(self, *args, currency=None, **kwargs):  # pylint: disable=arguments-differ
+    def wallet(self, currency=None, **kwargs):
```
See Issue #183 for the reason `*args` and `**kwargs` need to stay.
Moving the pylint comment above the line might actually be prettier.

firepol replied:
Ok, but *args must stay before currency or changing the order is ok? Because I was assuming, for consistency, that *args and **kwargs ideally should always be the last parameters of the methods. Not put something in the middle like the currency in this case...

firepol committed:
Fix several bittrex API issues

Fix withdraw & *_history methods (they are private: authenticate=True).
Add pair check in the trade_history method.
Various DRY-ups and clean-ups.

firepol commented:
Updated based on your feedback, just `def wallet(self, *args, currency=None, **kwargs)` doesn't convince me enough. I'd like to have *args and **kwargs at the end of the method signature, like this:

`def wallet(self, currency=None, *args, **kwargs)`

Any reason to put `currency` between them? It just makes it look inconsistent.

deepbrook commented:
Regarding the order of arguments:
It's a code-style thing - first come positionals, then keyword arguments.

nlsdfnbch commented on bitex/interface/bittrex.py:
```diff
@@ -42,49 +42,43 @@ def _get_supported_pairs(self):
-        payload.update(kwargs)
-        return self.request('public/getorderbook', params=payload)
+        kwargs.update({'market': pair, 'type': 'both'})
+        return self.request('public/getorderbook', params=kwargs)
```
This may not work - since this includes ALL kwargs passed, it would also send the ones not supported by bittrex, wich is going to result in a faulty request.
